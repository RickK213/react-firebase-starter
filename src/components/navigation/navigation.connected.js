import { connect } from 'react-redux';
import { Navigation } from './navigation';
import { selectAuthUser } from '../../store/auth-user/auth-user';

const mapStateToProps = state => ({
  authUser: selectAuthUser(state)
});

export const ConnectedNavigation = connect(mapStateToProps)(Navigation);
