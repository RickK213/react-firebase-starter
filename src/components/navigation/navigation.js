import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { ROUTES } from '../../constants/routes';
import { SignOutButton } from '../screens/auth-screens/sign-out-button/sign-out-button';

export const Navigation = props => {
  const { authUser } = props;

  return (
    <nav>
      <ul>
        <li>
          <Link to={ROUTES.HOME.path}>{ROUTES.HOME.label}</Link>
        </li>
        {authUser ? (
          <React.Fragment>
            <li>
              <Link to={ROUTES.ACCOUNT.path}>{ROUTES.ACCOUNT.label}</Link>
            </li>
            <li>
              <SignOutButton />
            </li>
          </React.Fragment>
        ) : (
          <React.Fragment>
            <li>
              <Link to={ROUTES.SIGN_UP.path}>{ROUTES.SIGN_UP.label}</Link>
            </li>
            <li>
              <Link to={ROUTES.SIGN_IN.path}>{ROUTES.SIGN_IN.label}</Link>
            </li>
          </React.Fragment>
        )}
      </ul>
    </nav>
  );
};

Navigation.propTypes = {
  authUser: PropTypes.object
};

Navigation.defaultProps = {
  authUser: null
};
