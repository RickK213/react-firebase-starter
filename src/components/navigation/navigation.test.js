/* global React */
import ShallowRenderer from 'react-test-renderer/shallow';
import { createMemoryHistory } from 'history';
import { Navigation } from './navigation';

describe('<Navigation />', () => {
  const renderer = new ShallowRenderer();
  const history = createMemoryHistory({ keyLength: 0 });

  it('should render correctly', () => {
    renderer.render(<Navigation history={history} />);
    const component = renderer.getRenderOutput();
    expect(component).toMatchSnapshot();
  });
});
