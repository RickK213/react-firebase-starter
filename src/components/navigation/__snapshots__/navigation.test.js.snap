// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render correctly 1`] = `
<nav>
  <ul>
    <li>
      <Link
        to="/"
      >
        Home
      </Link>
    </li>
    <li>
      <Link
        to="/sign-up"
      >
        Sign Up
      </Link>
    </li>
    <li>
      <Link
        to="/account"
      >
        Account
      </Link>
    </li>
  </ul>
</nav>
`;
